test/integration/scheduler/utils.go

```diff
@@ -0,0 +1,333 @@
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/integration/framework"
+
+	"net/http/httptest"
+)
+
+type TestContext struct {
+	closeFn                framework.CloseFunc
+	httpServer             *httptest.Server
+	ns                     *v1.Namespace
+	clientSet              *clientset.Clientset
+	informerFactory        informers.SharedInformerFactory
+	schedulerConfigFactory scheduler.Configurator
+	schedulerConfig        *scheduler.Config
+	scheduler              *scheduler.Scheduler
+}
+
+// initTest initializes a test environment and creates a scheduler with default
+// configuration.
+func initTest(t *testing.T, name string) *TestContext {
+	var context TestContext
+	_, context.httpServer, context.closeFn = framework.RunAMaster(nil)
+
+	context.ns = framework.CreateTestingNamespace(name, context.httpServer, t)
+
+	context.clientSet = clientset.NewForConfigOrDie(&restclient.Config{Host: context.httpServer.URL, ContentConfig: restclient.ContentConfig{GroupVersion: testapi.Groups[v1.GroupName].GroupVersion()}})
+	context.informerFactory = informers.NewSharedInformerFactory(context.clientSet, 0)
+
+	context.schedulerConfigFactory = factory.NewConfigFactory(
+		v1.DefaultSchedulerName,
+		context.clientSet,
+		context.informerFactory.Core().V1().Nodes(),
+		context.informerFactory.Core().V1().Pods(),
+		context.informerFactory.Core().V1().PersistentVolumes(),
+		context.informerFactory.Core().V1().PersistentVolumeClaims(),
+		context.informerFactory.Core().V1().ReplicationControllers(),
+		context.informerFactory.Extensions().V1beta1().ReplicaSets(),
+		context.informerFactory.Apps().V1beta1().StatefulSets(),
+		context.informerFactory.Core().V1().Services(),
+		v1.DefaultHardPodAffinitySymmetricWeight,
+	)
+	var err error
+	context.schedulerConfig, err = context.schedulerConfigFactory.Create()
+	if err != nil {
+		t.Fatalf("Couldn't create scheduler config: %v", err)
+	}
+	eventBroadcaster := record.NewBroadcaster()
+	context.schedulerConfig.Recorder = eventBroadcaster.NewRecorder(api.Scheme, clientv1.EventSource{Component: v1.DefaultSchedulerName})
+	eventBroadcaster.StartRecordingToSink(&clientv1core.EventSinkImpl{Interface: clientv1core.New(context.clientSet.CoreV1().RESTClient()).Events("")})
+	context.informerFactory.Start(context.schedulerConfig.StopEverything)
+	context.scheduler, err = scheduler.NewFromConfigurator(&scheduler.FakeConfigurator{Config: context.schedulerConfig}, nil...)
+	if err != nil {
+		t.Fatalf("Couldn't create scheduler: %v", err)
+	}
+	context.scheduler.Run()
+	return &context
+}
+
+// cleanupTest deletes the scheduler and the test namespace. It should be called
+// at the end of a test.
+func cleanupTest(t *testing.T, context *TestContext) {
+	// Kill the scheduler.
+	close(context.schedulerConfig.StopEverything)
+	// Cleanup nodes.
+	context.clientSet.CoreV1().Nodes().DeleteCollection(nil, metav1.ListOptions{})
+	framework.DeleteTestingNamespace(context.ns, context.httpServer, t)
```

Comment:
I think you are confusing this with our e2e tests. I don't remember that deleting the namespace has been a source of problem in integration tests. What caused trouble in e2e tests was not deletion of the namespace, it was the fact that many of our priority and predicate tests were failing due to inconsistencies in our test clusters. Those clusters are not predictable enough for these kinds of tests.